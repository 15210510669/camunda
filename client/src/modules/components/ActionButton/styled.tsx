/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import styled, {css} from 'styled-components';
import {IconButton} from 'modules/components/IconButton';

const Button = styled(IconButton)`
  ${({theme}) => {
    const colors = theme.colors.variables.editButton;

    return css`
      margin-left: 8px;
      z-index: 0;

      svg {
        margin-top: 4px;
      }

      &:disabled,
      &:disabled :hover {
        svg {
          color: ${colors.disabled.color};
          opacity: 0.5;
        }

        &:before {
          background-color: transparent;
        }
      }
    `;
  }}
`;

export {Button};
