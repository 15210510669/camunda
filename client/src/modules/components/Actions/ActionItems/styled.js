import styled, {css} from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';

import {ReactComponent as Retry} from 'modules/components/Icon/retry.svg';
import {ReactComponent as Stop} from 'modules/components/Icon/stop.svg';

export const Ul = themed(styled.ul`
  display: inline-flex;
  flex-direction: row;
  padding: 2px;
  margin: 0px;

  list-style-type: none;
  border-radius: 12px;

  background: ${({children}) =>
    // checks if children are components
    children.reduce((sum, next) => sum || !!next, false) &&
    themeStyle({
      dark: Colors.uiDark01,
      light: 'rgba(98, 98, 110, 0.15)'
    })};

  /* Border between single action items */
  & > li {
    &:not(:last-child) {
      border-right: solid 1px
        ${themeStyle({
          dark: 'rgba(250, 250, 250, 0.15)',
          light: 'rgb(98, 98, 110, 0.15)'
        })};
    }
  }

  /* Button border radius for correct focus style shapes*/

  /* only one child exists */
  & > li:first-child:last-child > button {
    border-radius: 12px;
  }

  & > li:first-child > button {
    border-radius: 12px 0 0 12px;
  }
  & > li:last-child > button {
    border-radius: 0 12px 12px 0;
  }
  & > li:not(:first-child):not(:last-child) > button {
    border-radius: 0px;
  }
`);

export const Button = themed(styled.button`
  display: flex;
  align-items: center;
  padding: 3px;

  background: none;
  border: none;
  border-radius: 12px;

  &:focus {
    outline: none;
    box-shadow: ${themeStyle({
      dark: `0 0 0 1px ${Colors.lightFocusInner},0 0 0 4px ${
        Colors.focusOuter
      }`,

      light: `0 0 0 1px ${Colors.darkFocusInner}, 0 0 0 4px ${
        Colors.focusOuter
      }`
    })};
  }
`);

export const iconStyle = css`
  opacity: 0.7;
  color: ${themeStyle({
    dark: '#ffffff',
    light: Colors.uiDark02
  })};
  cursor: pointer;
`;

// import and style any Icon accordingly
export const RetryIcon = themed(styled(Retry)`
  ${iconStyle};
`);

export const CancelIcon = themed(styled(Stop)`
  ${iconStyle};
`);
