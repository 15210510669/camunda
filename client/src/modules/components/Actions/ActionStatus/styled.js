import styled from 'styled-components';

import Spinner from 'modules/components/Spinner';

import {Colors, themed, themeStyle} from 'modules/theme';

export const ActionSpinner = themed(styled(Spinner)`
  margin: 0 5px;
 border: 3px solid ${({selected}) =>
   themeStyle({
     dark: '#ffffff',
     light: selected ? Colors.selections : Colors.badge02
   })};
    border-right-color: transparent;
  }
`);
