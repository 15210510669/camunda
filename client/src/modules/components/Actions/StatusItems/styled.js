import styled, {css} from 'styled-components';
import {Colors, themed} from 'modules/theme';

import {ReactComponent as Retry} from 'modules/components/Icon/retry.svg';
import {ReactComponent as Stop} from 'modules/components/Icon/retry.svg';

export const Ul = themed(styled.ul`
  Â display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 0 0 2px;
  margin: 0px;

  list-style-type: none;
  border-radius: 12px;

  background: transparent;
`);

export const Li = themed(styled.li`
  display: flex;
  align-items: center;
  padding: 3px;

  background: none;
  border: none;
  border-radius: 12px;
`);

export const iconStyle = css`
  color: ${Colors.incidentsAndErrors};
`;

// import and style any Icon accordingly
export const RetryIcon = themed(styled(Retry)`
  ${iconStyle};
`);

export const CancelIcon = themed(styled(Stop)`
  ${iconStyle};
`);
