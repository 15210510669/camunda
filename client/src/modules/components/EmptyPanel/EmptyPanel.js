/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled';

function EmptyPanel({type, label, ...props}) {
  return (
    <Styled.EmptyPanel {...props}>
      <Styled.LabelContainer>
        {type === 'warning' && <Styled.WarningIcon />}
        <Styled.Label type={type}>{label}</Styled.Label>
      </Styled.LabelContainer>
    </Styled.EmptyPanel>
  );
}

EmptyPanel.propTypes = {
  label: PropTypes.string,
  type: PropTypes.oneOf(['info', 'warning'])
};

export default EmptyPanel;
