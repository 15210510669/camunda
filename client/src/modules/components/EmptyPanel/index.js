/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {useRef, useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled';

export default function EmptyPanel({
  type,
  label,
  rowHeight,
  Skeleton,
  ...props
}) {
  const containerRef = useRef(null);

  return (
    <Styled.EmptyPanel {...props} ref={containerRef}>
      {type === 'skeleton' ? (
        <WithRowCount rowHeight={rowHeight} containerRef={containerRef}>
          <Skeleton />
        </WithRowCount>
      ) : (
        <Styled.LabelContainer>
          {type === 'warning' && (
            <Styled.WarningIcon data-testid="warning-icon" />
          )}
          <Styled.Label type={type}>{label}</Styled.Label>
        </Styled.LabelContainer>
      )}
    </Styled.EmptyPanel>
  );
}

EmptyPanel.propTypes = {
  label: PropTypes.string,
  type: PropTypes.oneOf(['info', 'warning', 'skeleton']),
  rowHeight: PropTypes.number,
  Skeleton: PropTypes.elementType,
};

/**
 * @returns {number} the number of rows a skelton should show, based on the available screen height and the row height of the data expected to be shown
 * @param {number} rowHeight row height of the data expected to be shown
 * @param {obj} containerRef ref of the parent component
 */
export const WithRowCount = function ({rowHeight, containerRef, ...props}) {
  const [rows, setRows] = useState(0);

  useEffect(() => {
    function recalculateHeight() {
      if (containerRef.current) {
        const rows = Math.floor(containerRef.current.clientHeight / rowHeight);

        setRows(rows);
      }
    }

    recalculateHeight();

    window.addEventListener('resize', recalculateHeight);
    return () => window.removeEventListener('resize', recalculateHeight);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  function renderChildren() {
    return React.Children.map(
      props.children,
      (child) =>
        child &&
        React.cloneElement(child, {
          rowsToDisplay: rows,
        })
    );
  }

  return renderChildren();
};
