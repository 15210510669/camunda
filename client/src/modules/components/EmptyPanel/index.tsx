/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React, {useRef, useState, useEffect} from 'react';
import * as Styled from './styled';

type Props = {
  label?: string;
  type?: 'info' | 'warning' | 'skeleton';
  rowHeight?: number;
  Skeleton?: React.FC;
};

const EmptyPanel: React.FC<Props> = ({
  type,
  label,
  rowHeight,
  Skeleton,
  ...props
}) => {
  const containerRef = useRef(null);

  return (
    <Styled.EmptyPanel {...props} ref={containerRef}>
      {type === 'skeleton' && Skeleton !== undefined ? (
        <WithRowCount rowHeight={rowHeight} containerRef={containerRef}>
          <Skeleton />
        </WithRowCount>
      ) : (
        <Styled.LabelContainer>
          {type === 'warning' && (
            <Styled.WarningIcon data-testid="warning-icon" />
          )}
          {/* @ts-expect-error ts-migrate(2769) FIXME: Property 'type' does not exist on type 'IntrinsicA... Remove this comment to see the full error message */}
          <Styled.Label type={type}>{label}</Styled.Label>
        </Styled.LabelContainer>
      )}
    </Styled.EmptyPanel>
  );
};

/**
 * @returns {number} the number of rows a skelton should show, based on the available screen height and the row height of the data expected to be shown
 * @param {number} rowHeight row height of the data expected to be shown
 * @param {obj} containerRef ref of the parent component
 */
export const WithRowCount = function ({
  rowHeight,
  containerRef,
  ...props
}: any) {
  const [rows, setRows] = useState(0);

  useEffect(() => {
    function recalculateHeight() {
      if (containerRef.current) {
        const rows = Math.floor(containerRef.current.clientHeight / rowHeight);

        setRows(rows);
      }
    }

    recalculateHeight();

    window.addEventListener('resize', recalculateHeight);
    return () => window.removeEventListener('resize', recalculateHeight);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  function renderChildren() {
    return React.Children.map(
      props.children,
      (child) =>
        child &&
        React.cloneElement(child, {
          rowsToDisplay: rows,
        }),
    );
  }

  return renderChildren();
};

export {EmptyPanel};
