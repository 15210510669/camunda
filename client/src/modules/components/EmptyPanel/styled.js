/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import styled, {css} from 'styled-components';

import {ReactComponent as Check} from 'modules/components/Icon/check.svg';
import {ReactComponent as Warning} from 'modules/components/Icon/warning-message-icon.svg';

const EmptyPanel = styled.div`
  height: 100%;
  width: 100%;
`;

const LabelContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 58%;
`;

const Label = styled.span`
  ${({theme, type}) => {
    const colors = theme.colors.modules.emptyPanel;

    return css`
      ${type === 'info'
        ? css`
            color: ${colors.color};
            opacity: 0.8;
          `
        : css`
            color: ${theme.colors.incidentsAndErrors};
            opacity: 0.9;
          `}

      font-family: IBMPlexSans;
      font-size: 16px;
      padding-top: 5px;
    `;
  }}
`;

const CheckIcon = styled(Check)`
  ${({theme}) => {
    return css`
      width: 18px;
      height: 14px;
      fill: ${theme.colors.allIsWell};
      margin-right: 13px;
    `;
  }}
`;
const WarningIcon = styled(Warning)`
  ${({theme}) => {
    return css`
      width: 20px;
      height: 18px;
      fill: ${theme.colors.incidentsAndErrors};
      margin-right: 15px;
    `;
  }}
`;

export {EmptyPanel, LabelContainer, Label, CheckIcon, WarningIcon};
