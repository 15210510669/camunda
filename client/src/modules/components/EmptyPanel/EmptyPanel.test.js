/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {mount} from 'enzyme';
import EmptyPanel, {WithRowCount} from './EmptyPanel';
import * as Styled from './styled';

const label = 'someLabel';
const SkeletonMock = () => <div data-test="Skeleton"></div>;

describe('EmptyPanel', () => {
  it('should display a warning message', () => {
    const node = mount(<EmptyPanel label={label} type="warning" />);

    expect(node.text()).toContain(label);
    expect(node.find(Styled.WarningIcon)).toExist();
  });

  it('should display a success message', () => {
    const node = mount(<EmptyPanel label={label} type="info" />);
    expect(node.text()).toContain(label);
  });

  it('should display a skeleton', () => {
    const node = mount(
      <EmptyPanel label={label} type="skeleton" Skeleton={SkeletonMock} />
    );

    expect(node.find(SkeletonMock)).toExist();
  });

  describe('WithRowCount', () => {
    it('should calculate number of shown skeleton rows', () => {
      const containerRef = {current: {clientHeight: 200}};

      const node = mount(
        <WithRowCount rowHeight={12} containerRef={containerRef}>
          <SkeletonMock />
        </WithRowCount>
      );

      expect(node.find(SkeletonMock).props().rowsToDisplay).toBe(22);
    });
  });
});
