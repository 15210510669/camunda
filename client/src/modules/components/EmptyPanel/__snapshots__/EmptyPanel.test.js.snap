// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyPanel should display a skeleton 1`] = `
<EmptyPanel
  Skeleton={<div />}
  label="someLabel"
  type="info"
>
  <Themed(styled.div)>
    <Themed
      forwardedRef={
        Object {
          "current": <div
            class="sc-bdVaJa imdMaR"
          >
            <div
              class="sc-bwzfXH kbFnXb"
            >
              <span
                class="sc-htpNat heVysk"
                type="info"
              >
                someLabel
              </span>
            </div>
          </div>,
        }
      }
    >
      <styled.div
        theme="dark"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "imdMaR",
                "rules": Array [
                  "
  height: 100%;
  width: 100%;

  /* border-width: 1px; */
  /* border-style: solid; */
  /* border-color: ",
                  [Function],
                  "; */
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={
            Object {
              "current": <div
                class="sc-bdVaJa imdMaR"
              >
                <div
                  class="sc-bwzfXH kbFnXb"
                >
                  <span
                    class="sc-htpNat heVysk"
                    type="info"
                  >
                    someLabel
                  </span>
                </div>
              </div>,
            }
          }
          theme="dark"
        >
          <div
            className="sc-bdVaJa imdMaR"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "kbFnXb",
                      "rules": Array [
                        "
  display: flex;
  align-items: center;
  justify-content: center;
  height: 58%;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-bwzfXH kbFnXb"
                >
                  <Themed(styled.span)
                    type="info"
                  >
                    <Themed
                      forwardedRef={null}
                      type="info"
                    >
                      <styled.span
                        theme="dark"
                        type="info"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": false,
                                "lastClassName": "heVysk",
                                "rules": Array [
                                  "
  ",
                                  [Function],
                                  "

  font-family: IBMPlexSans;
  font-size: 16px;
  padding-top: 5px;
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          theme="dark"
                          type="info"
                        >
                          <span
                            className="sc-htpNat heVysk"
                            type="info"
                          >
                            someLabel
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </Themed>
                  </Themed(styled.span)>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Themed>
  </Themed(styled.div)>
</EmptyPanel>
`;

exports[`EmptyPanel should display a success message 1`] = `
<EmptyPanel
  label="someLabel"
  type="info"
>
  <Themed(styled.div)>
    <Themed
      forwardedRef={
        Object {
          "current": <div
            class="sc-bdVaJa imdMaR"
          >
            <div
              class="sc-bwzfXH kbFnXb"
            >
              <span
                class="sc-htpNat heVysk"
                type="info"
              >
                someLabel
              </span>
            </div>
          </div>,
        }
      }
    >
      <styled.div
        theme="dark"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "imdMaR",
                "rules": Array [
                  "
  height: 100%;
  width: 100%;

  /* border-width: 1px; */
  /* border-style: solid; */
  /* border-color: ",
                  [Function],
                  "; */
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={
            Object {
              "current": <div
                class="sc-bdVaJa imdMaR"
              >
                <div
                  class="sc-bwzfXH kbFnXb"
                >
                  <span
                    class="sc-htpNat heVysk"
                    type="info"
                  >
                    someLabel
                  </span>
                </div>
              </div>,
            }
          }
          theme="dark"
        >
          <div
            className="sc-bdVaJa imdMaR"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "kbFnXb",
                      "rules": Array [
                        "
  display: flex;
  align-items: center;
  justify-content: center;
  height: 58%;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-bwzfXH kbFnXb"
                >
                  <Themed(styled.span)
                    type="info"
                  >
                    <Themed
                      forwardedRef={null}
                      type="info"
                    >
                      <styled.span
                        theme="dark"
                        type="info"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": false,
                                "lastClassName": "heVysk",
                                "rules": Array [
                                  "
  ",
                                  [Function],
                                  "

  font-family: IBMPlexSans;
  font-size: 16px;
  padding-top: 5px;
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          theme="dark"
                          type="info"
                        >
                          <span
                            className="sc-htpNat heVysk"
                            type="info"
                          >
                            someLabel
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </Themed>
                  </Themed(styled.span)>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Themed>
  </Themed(styled.div)>
</EmptyPanel>
`;

exports[`EmptyPanel should display a warning message 1`] = `
<EmptyPanel
  label="someLabel"
  type="warning"
>
  <Themed(styled.div)>
    <Themed
      forwardedRef={
        Object {
          "current": <div
            class="sc-bdVaJa imdMaR"
          >
            <div
              class="sc-bwzfXH kbFnXb"
            >
              <svg
                class="sc-ifAKCX cZHWPV"
              >
                warning-message-icon.svg
              </svg>
              <span
                class="sc-htpNat hlJWtx"
                type="warning"
              >
                someLabel
              </span>
            </div>
          </div>,
        }
      }
    >
      <styled.div
        theme="dark"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "imdMaR",
                "rules": Array [
                  "
  height: 100%;
  width: 100%;

  /* border-width: 1px; */
  /* border-style: solid; */
  /* border-color: ",
                  [Function],
                  "; */
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={
            Object {
              "current": <div
                class="sc-bdVaJa imdMaR"
              >
                <div
                  class="sc-bwzfXH kbFnXb"
                >
                  <svg
                    class="sc-ifAKCX cZHWPV"
                  >
                    warning-message-icon.svg
                  </svg>
                  <span
                    class="sc-htpNat hlJWtx"
                    type="warning"
                  >
                    someLabel
                  </span>
                </div>
              </div>,
            }
          }
          theme="dark"
        >
          <div
            className="sc-bdVaJa imdMaR"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "kbFnXb",
                      "rules": Array [
                        "
  display: flex;
  align-items: center;
  justify-content: center;
  height: 58%;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-bwzfXH kbFnXb"
                >
                  <Styled(ReactComponent)>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "cZHWPV",
                            "rules": Array [
                              "
  width: 20px;
  height: 18px;
  fill: ",
                              "#ff3d3d",
                              ";
  margin-right: 15px;
",
                            ],
                          },
                          "displayName": "Styled(ReactComponent)",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": [Function],
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                          Symbol(Symbol.hasInstance): [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <ReactComponent
                        className="sc-ifAKCX cZHWPV"
                      >
                        <svg
                          className="sc-ifAKCX cZHWPV"
                        >
                          warning-message-icon.svg
                        </svg>
                      </ReactComponent>
                    </StyledComponent>
                  </Styled(ReactComponent)>
                  <Themed(styled.span)
                    type="warning"
                  >
                    <Themed
                      forwardedRef={null}
                      type="warning"
                    >
                      <styled.span
                        theme="dark"
                        type="warning"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": false,
                                "lastClassName": "hlJWtx",
                                "rules": Array [
                                  "
  ",
                                  [Function],
                                  "

  font-family: IBMPlexSans;
  font-size: 16px;
  padding-top: 5px;
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          theme="dark"
                          type="warning"
                        >
                          <span
                            className="sc-htpNat hlJWtx"
                            type="warning"
                          >
                            someLabel
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </Themed>
                  </Themed(styled.span)>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Themed>
  </Themed(styled.div)>
</EmptyPanel>
`;
